source/gmsh/utilities.cc

```diff
@@ -144,15 +155,201 @@ namespace Gmsh
   }
 #  endif
 
+#  ifdef DEAL_II_GMSH_WITH_API
+  template <int dim, int spacedim>
+  void
+  read_partitioned_msh(
+    parallel::fullydistributed::Triangulation<dim, spacedim> &tria,
+    const MPI_Comm                                            mpi_comm,
+    const std::string                                        &file_prefix,
+    const std::string                                        &file_suffix)
+  {
+    const unsigned int nprocs = Utilities::MPI::n_mpi_processes(mpi_comm);
+    const unsigned int rank   = Utilities::MPI::this_mpi_process(mpi_comm);
+
+    std::string fname =
+      file_prefix + "_" + std::to_string(rank + 1) + "." + file_suffix;
+
+    if (nprocs == 1)
+      {
+        fname = file_prefix + "." + file_suffix;
+        std::ifstream f(fname);
+        AssertThrow(f.good(), ExcMessage("Missing mesh file: " + fname));
```

Comment:
You can do this more elegantly using this function: https://en.cppreference.com/w/cpp/filesystem/exists.html